### PR TITLE
Polish tests

### PR DESCRIPTION
**Pull request by quaff**
1. Improve null safety
2. Replace lambda with method reference

### CODE REVIEW

File: spring-boot-project/spring-boot/src/test/java/org/springframework/boot/context/properties/ConfigurationPropertiesBeanTests.java [hunk 1]
```diff
@@ -197,6 +197,7 @@ void getWhenHasFactoryMethodAndBeanAnnotationFavorsFactoryMethod() throws Throwa
 	void getWhenHasValidatedBeanBindsWithBeanAnnotation() throws Throwable {
 		get(ValidatedBeanConfiguration.class, "validatedBean", (propertiesBean) -> {
 			Validated validated = propertiesBean.asBindTarget().getAnnotation(Validated.class);
+			assertThat(validated).isNotNull();
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot/src/test/java/org/springframework/boot/context/properties/ConfigurationPropertiesBeanTests.java:200]: I don't think this is an improvement. It makes the test more verbose and I think the benefit is negligible. Should `validated` be `null`, the NPE that would occur on the next line when calling `validated.value()` will make that clear.

File: spring-boot-project/spring-boot/src/test/java/org/springframework/boot/context/properties/ConfigurationPropertiesBeanTests.java [hunk 2]
```diff
@@ -205,6 +206,7 @@ void getWhenHasValidatedBeanBindsWithBeanAnnotation() throws Throwable {
 	void getWhenHasValidatedFactoryMethodBindsWithFactoryMethodAnnotation() throws Throwable {
 		get(ValidatedMethodConfiguration.class, "annotatedBean", (propertiesBean) -> {
 			Validated validated = propertiesBean.asBindTarget().getAnnotation(Validated.class);
+			assertThat(validated).isNotNull();
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot/src/test/java/org/springframework/boot/context/properties/ConfigurationPropertiesBeanTests.java:209]: As above. Please revert.

File: spring-boot-project/spring-boot/src/test/java/org/springframework/boot/context/properties/ConfigurationPropertiesBeanTests.java [hunk 3]
```diff
@@ -213,6 +215,7 @@ void getWhenHasValidatedFactoryMethodBindsWithFactoryMethodAnnotation() throws T
 	void getWhenHasValidatedBeanAndFactoryMethodBindsWithFactoryMethodAnnotation() throws Throwable {
 		get(ValidatedMethodAndBeanConfiguration.class, "validatedBean", (propertiesBean) -> {
 			Validated validated = propertiesBean.asBindTarget().getAnnotation(Validated.class);
+			assertThat(validated).isNotNull();
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot/src/test/java/org/springframework/boot/context/properties/ConfigurationPropertiesBeanTests.java:218]: As above. Please revert.

File: spring-boot-project/spring-boot/src/test/java/org/springframework/boot/context/properties/ConfigurationPropertiesBindExceptionTests.java [hunk 4]
```diff
@@ -36,6 +36,7 @@ void createFromBeanHasDetails() {
 		ApplicationContext applicationContext = new AnnotationConfigApplicationContext(Example.class);
 		ConfigurationPropertiesBean bean = ConfigurationPropertiesBean.get(applicationContext,
 				applicationContext.getBean(Example.class), "example");
+		assertThat(bean).isNotNull();
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot/src/test/java/org/springframework/boot/context/properties/ConfigurationPropertiesBindExceptionTests.java:39]: As above. Please revert. Additionally, `getBean(Class)` will never return `null`, throwing a `NoSuchBeanDefinitionException` instead.

File: spring-boot-project/spring-boot/src/test/java/org/springframework/boot/context/properties/ConfigurationPropertiesTests.java [hunk 5]
```diff
@@ -1197,6 +1197,7 @@ void loadWhenBoundToRandomPropertyPlaceholder() {
 	void boundPropertiesShouldBeRecorded() {
 		load(NestedConfiguration.class, "name=foo", "nested.name=bar");
 		BoundConfigurationProperties bound = BoundConfigurationProperties.get(this.context);
+		assertThat(bound).isNotNull();
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot/src/test/java/org/springframework/boot/context/properties/ConfigurationPropertiesTests.java:1200]: As above. Please revert.

**Review by wilkinsona — CHANGES_REQUESTED** [review]

File: spring-boot-project/spring-boot/src/test/java/org/springframework/boot/context/properties/ConfigurationPropertiesBindExceptionTests.java (hunk 4 above)
Comment (quaff) [reply spring-boot-project/spring-boot/src/test/java/org/springframework/boot/context/properties/ConfigurationPropertiesBindExceptionTests.java:39]: `ConfigurationPropertiesBean.get()` is `@Nullable`

File: spring-boot-project/spring-boot/src/test/java/org/springframework/boot/context/properties/ConfigurationPropertiesBeanTests.java (hunk 1 above)
Comment (quaff) [reply spring-boot-project/spring-boot/src/test/java/org/springframework/boot/context/properties/ConfigurationPropertiesBeanTests.java:200]: It makes IDEA and code analysis tools happy, is it worthy?

Comment (wilkinsona) [reply spring-boot-project/spring-boot/src/test/java/org/springframework/boot/context/properties/ConfigurationPropertiesBeanTests.java:200]: No, I don't think so.

File: spring-boot-project/spring-boot/src/test/java/org/springframework/boot/context/properties/ConfigurationPropertiesBindExceptionTests.java (hunk 4 above)
Comment (wilkinsona) [reply spring-boot-project/spring-boot/src/test/java/org/springframework/boot/context/properties/ConfigurationPropertiesBindExceptionTests.java:39]: Sorry, I misread the diff. Regardless, the same reasoning as above still applies. I don't think null checks such as this in tests are beneficial.

**Comment by wilkinsona** [comment]
Thanks, @quaff.

I've merged this without the null checks. We can consider this topic again as and when we tackle https://github.com/spring-projects/spring-boot/issues/10712, building on the Framework team's experience where they treat main and test source differently.

**Pull request closed without merging**